compose.yaml

```diff
@@ -45,50 +38,78 @@ x-shopware: &shopware-65
     - cache:/var/www/html/var/cache
 
 services:
-    database:
-        <<: *service
-        image: mysql:8.0
-        entrypoint:
-            [
-                "sh",
-                "-c",
-                "docker-entrypoint.sh mysqld --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci --sql-require-primary-key=ON"
-            ]
-        tmpfs:
-            - /var/lib/mysql:uid=999,gid=999
+  database:
+    <<: [ *service ]
+    image: mysql:8.0
+    entrypoint:
+      [
+        "sh",
+        "-c",
+        "docker-entrypoint.sh mysqld --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci --sql-require-primary-key=ON"
+      ]
+    tmpfs:
+      - /var/lib/mysql:uid=999,gid=999
+    healthcheck:
+      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
+      start_interval: 1s
+      start_period: 10s
 
-    adminer:
-        <<: *service
-        image: adminer:latest
-        depends_on: [ database ]
-        ports:
-            - '8012:8080'
+  adminer:
+    <<: [ *service ]
+    image: adminer:latest
+    depends_on:
+      database:
+        condition: service_healthy
+    ports:
+      - '8012:8080'
 
-    mailpit:
-        <<: *service
-        image: axllent/mailpit
-        ports:
-            - 8013:8025
-    shopware:
-        <<: *service
-        image: ghcr.io/shopware/shopware/ci-e2e:trunk
-        depends_on: [ database, mailpit ]
-        ports:
-            - '8011:8000'
+  mailpit:
+    <<: [ *service ]
+    image: axllent/mailpit
+    ports:
+      - 8013:8025
+    healthcheck:
+      test: ["CMD", "wget", "-q", "--spider", "http://localhost:8025"]
 
-    init-65:
-        <<: *shopware-65
-        depends_on: [ database, mailpit ]
-        entrypoint: ['sh', '/init.sh']
-        user: root
+  init-volumes:
+    <<: [ *shopware-volumes ]
+    image: alpine:latest
+    entrypoint: ["chown"]
+    command: ["-R", "82:82", "/var/www/html"]
+    user: root
 
-    shopware-65:
-        <<: *shopware-65
-        depends_on:
-            init-65:
-                condition: service_completed_successfully
-        ports:
-            - '8021:8000'
+  init:
+    <<: [ *service, *shopware-volumes ]
+    image: ghcr.io/shopware/acceptance-test-suite/test-image:${SHOPWARE_VERSION:-trunk}
+    build:
+      context: .
+      dockerfile: test-image.Dockerfile
+      args:
+        - PHP_VERSION=${PHP_VERSION:-8.4}
+        - SHOPWARE_VERSION=${SHOPWARE_VERSION:-trunk}
+        - SHOPWARE_BUILD_SOURCE=${SHOPWARE_BUILD_SOURCE:-branch}
```

Comment:
```suggestion
```
The images are build in the test jobs right. We should only build them in the nightly

Reply:
This is useful for regular usage of the `compose.yaml`, so I'd like to keep it; once the images have been registered/pushed by the nightly once, they will not be built anymore, but only pulled by the test job.